hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -947,7 +947,14 @@ public String getTypeName(SqlTypeDescriptor sqlTypeDescriptor) throws HibernateE
 	}
 
 	public String getTypeName(int code) throws HibernateException {
-		return getTypeName( code, Size.nil() );
+		switch ( code ) {
+			case Types.TIMESTAMP:
+			case Types.TIMESTAMP_WITH_TIMEZONE:
+				// explicitly enforce dialect's default timestamp precision
+				return getTypeName( code, Size.precision( getDefaultTimestampPrecision() ) );
```

Comment:
Shouldn't we then also use the default precision for `DECIMAL`/`NUMERIC` (`Dialect.getDefaultDecimalPrecision`), `FLOAT`/`REAL` (`Dialect.getFloatPrecision`) and `DOUBLE` (`Dialect.getDoublePrecision`)?

Reply:
Quite possibly.  I can certainly add those here but the Envers tests don't have test cases for those tmk.